Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp

```diff
@@ -242,18 +372,22 @@ class ScriptMessageClientGtk final : public WebScriptMessageHandler::Client {
 
     bool supportsAsyncReply() override
     {
-        return false;
+        return m_supportsAsyncReply;
     }
-    
-    void didPostMessageWithAsyncReply(WebPageProxy&, FrameInfoData&&, API::ContentWorld&, WebCore::SerializedScriptValue&, WTF::Function<void(API::SerializedScriptValue*, const String&)>&&) override
+
+    void didPostMessageWithAsyncReply(WebPageProxy&, FrameInfoData&&, API::ContentWorld&, WebCore::SerializedScriptValue& serializedScriptValue, WTF::Function<void(API::SerializedScriptValue*, const String&)>&& completionHandler) override
     {
+        WebKitUserScriptMessage* message = webKitUserScriptMessageCreate(serializedScriptValue, completionHandler);
+        g_signal_emit(m_manager, signals[SCRIPT_REQUEST_RECEIVED], m_handlerName, message);
+        webkit_javascript_result_unref(message);
```

Comment:
This is using the incorrect unref function for `WebKitUserScriptMessage`, should be:

```suggestion
        webkit_user_script_message_unref(message);
```